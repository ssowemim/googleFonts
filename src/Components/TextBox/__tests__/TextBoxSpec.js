import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { TextBox } from './../index';

describe('TextBox', () => {
    let textBoxProps;
    beforeEach(()=>{
        textBoxProps = {
            fontFamily: "Oswald",
            fontVariant: "Regular",
            fontSize: "40",
            color: {
                r: '0',
                g: '0',
                b: '0',
                a: '1',
              }
        }
    })
    it('renders correctly', () => {
        let tree = renderer.create(
            <TextBox {...textBoxProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should toggle isHowToUseTabVisible on how to use btn click', ()=>{
        let wrapper = shallow(<TextBox {...textBoxProps} />);
        let elm = wrapper.find('.HowToUse');
        elm.simulate("click");
        expect(wrapper.instance().state.isHowToUseTabVisible).toBeTruthy()
        elm.simulate("click");
        expect(wrapper.instance().state.isHowToUseTabVisible).toBeFalsy()

    })
});
