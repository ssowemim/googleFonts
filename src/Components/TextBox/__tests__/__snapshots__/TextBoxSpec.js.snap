// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox renders correctly 1`] = `
<div
  className="undefined TextBox"
  onClick={[Function]}
>
  <div
    className="TextBox__TextArea"
  >
    <ul
      className="Texbx__FontDetails"
    >
      <li
        className="FontsDetails__Item FontsDetails__Item--FontFamily"
      >
        Oswald
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontVariant"
      >
        Regular
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontSize"
      >
        40
        px
      </li>
    </ul>
    <textarea
      autoFocus={undefined}
      placeholder="Write Something ..."
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "'Oswald', sans-serif",
          "fontSize": "40px",
          "fontStyle": "normal",
          "fontWeight": 400,
        }
      }
    />
  </div>
  <div
    className="TextBox__CTA"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-plus"
      />
      Compare
    </button>
    <button
      className="HowToUse"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-code"
      />
      How to use
    </button>
  </div>
</div>
`;
